## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#4230c15c-574a-433a-b456-f8614d63c598) |
| mint-large-bucket.sh | :x: [more...](#726b16a7-b612-4ea4-9a50-3aa430b669e1) |
| mint-fs.sh | :x: [more...](#29c4c515-ce50-4d4b-bbf4-5466cc102f24) |
| mint-gateway-s3.sh | :x: [more...](#7223c2e9-27fb-47ea-84a8-0aae6a597c7f) |
| mint-gateway-azure.sh | :x: [more...](#7ae87232-f3d9-4037-aa83-17b5b34b359e) |
<a name="4230c15c-574a-433a-b456-f8614d63c598"></a>
### 9598-99d6a88/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:30170
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c4177f5a518b:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 1 seconds
(5/15) Running awscli tests ... FAILED in 11 minutes and 14 seconds
{"name": "awscli", "duration": 625957, "function": "aws --endpoint-url http://minio-dev4.minio.io:30170 s3 cp /mint/data/datafile-65-MB s3://awscli-mint-test-bucket-6590/datafile-65-MB\n", "status": "FAIL", "error": "Completed 256.0 KiB/65.0 MiB (3.6 MiB/s) with 1 file(s) remaining
Completed 512.0 KiB/65.0 MiB (6.8 MiB/s) with 1 file(s) remaining
Completed 768.0 KiB/65.0 MiB (9.1 MiB/s) with 1 file(s) remaining
Completed 1.0 MiB/65.0 MiB (11.7 MiB/s) with 1 file(s) remaining 
Completed 1.2 MiB/65.0 MiB (9.4 MiB/s) with 1 file(s) remaining  
Completed 1.5 MiB/65.0 MiB (11.1 MiB/s) with 1 file(s) remaining 
Completed 1.8 MiB/65.0 MiB (12.8 MiB/s) with 1 file(s) remaining 
Completed 2.0 MiB/65.0 MiB (14.4 MiB/s) with 1 file(s) remaining 
Completed 2.2 MiB/65.0 MiB (15.9 MiB/s) with 1 file(s) remaining 
Completed 2.5 MiB/65.0 MiB (17.4 MiB/s) with 1 file(s) remaining 
Completed 2.8 MiB/65.0 MiB (18.9 MiB/s) with 1 file(s) remaining 
Completed 3.0 MiB/65.0 MiB (20.3 MiB/s) with 1 file(s) remaining 
Completed 3.2 MiB/65.0 MiB (21.7 MiB/s) with 1 file(s) remaining 
Completed 3.5 MiB/65.0 MiB (23.0 MiB/s) with 1 file(s) remaining 
Completed 3.8 MiB/65.0 MiB (24.3 MiB/s) with 1 file(s) remaining 
Completed 4.0 MiB/65.0 MiB (25.6 MiB/s) with 1 file(s) remaining 
Completed 4.2 MiB/65.0 MiB (26.9 MiB/s) with 1 file(s) remaining 
Completed 4.5 MiB/65.0 MiB (28.1 MiB/s) with 1 file(s) remaining 
Completed 4.8 MiB/65.0 MiB (29.2 MiB/s) with 1 file(s) remaining 
Completed 5.0 MiB/65.0 MiB (30.4 MiB/s) with 1 file(s) remaining 
Completed 5.2 MiB/65.0 MiB (31.5 MiB/s) with 1 file(s) remaining 
Completed 5.5 MiB/65.0 MiB (32.6 MiB/s) with 1 file(s) remaining 
Completed 5.8 MiB/65.0 MiB (33.7 MiB/s) with 1 file(s) remaining 
Completed 6.0 MiB/65.0 MiB (34.7 MiB/s) with 1 file(s) remaining 
Completed 6.2 MiB/65.0 MiB (35.7 MiB/s) with 1 file(s) remaining 
Completed 6.5 MiB/65.0 MiB (36.7 MiB/s) with 1 file(s) remaining 
Completed 6.8 MiB/65.0 MiB (37.7 MiB/s) with 1 file(s) remaining 
Completed 7.0 MiB/65.0 MiB (38.6 MiB/s) with 1 file(s) remaining 
Completed 7.2 MiB/65.0 MiB (39.5 MiB/s) with 1 file(s) remaining 
Completed 7.5 MiB/65.0 MiB (40.2 MiB/s) with 1 file(s) remaining 
Completed 7.8 MiB/65.0 MiB (41.2 MiB/s) with 1 file(s) remaining 
Completed 8.0 MiB/65.0 MiB (42.1 MiB/s) with 1 file(s) remaining 
Completed 8.2 MiB/65.0 MiB (42.9 MiB/s) with 1 file(s) remaining 
Completed 8.5 MiB/65.0 MiB (43.7 MiB/s) with 1 file(s) remaining 
Completed 8.8 MiB/65.0 MiB (44.6 MiB/s) with 1 file(s) remaining 
Completed 9.0 MiB/65.0 MiB (45.4 MiB/s) with 1 file(s) remaining 
Completed 9.2 MiB/65.0 MiB (151.7 KiB/s) with 1 file(s) remaining
Completed 9.5 MiB/65.0 MiB (155.3 KiB/s) with 1 file(s) remaining
Completed 9.8 MiB/65.0 MiB (159.3 KiB/s) with 1 file(s) remaining
Completed 10.0 MiB/65.0 MiB (162.3 KiB/s) with 1 file(s) remaining
Completed 10.2 MiB/65.0 MiB (166.2 KiB/s) with 1 file(s) remaining
Completed 10.5 MiB/65.0 MiB (170.2 KiB/s) with 1 file(s) remaining
Completed 10.8 MiB/65.0 MiB (173.5 KiB/s) with 1 file(s) remaining
Completed 10.8 MiB/65.0 MiB (35.8 KiB/s) with 1 file(s) remaining 
Completed 10.9 MiB/65.0 MiB (35.9 KiB/s) with 1 file(s) remaining 
Completed 10.9 MiB/65.0 MiB (36.0 KiB/s) with 1 file(s) remaining 
Completed 11.0 MiB/65.0 MiB (36.1 KiB/s) with 1 file(s) remaining 
Completed 11.0 MiB/65.0 MiB (36.2 KiB/s) with 1 file(s) remaining 
Completed 11.1 MiB/65.0 MiB (36.2 KiB/s) with 1 file(s) remaining 
Completed 11.2 MiB/65.0 MiB (36.3 KiB/s) with 1 file(s) remaining 
upload failed: ../../../data/datafile-65-MB to s3://awscli-mint-test-bucket-6590/datafile-65-MB Connection was closed before we received a valid response from endpoint URL: "http://minio-dev4.minio.io:30170/awscli-mint-test-bucket-6590/datafile-65-MB?uploadId=9b7db6c6-181b-4b68-a0c3-c5b40fbb8f95&partNumber=1"."}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... FAILED in 4 hours 1 minutes and 34 seconds
{
  "name": "mc",
  "duration": "7911440",
  "function": "test_mirror_list_objects",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-18858 --quiet --no-color mirror /mint/data myminio/mc-test-bucket-13776  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-13776/datafile-1-MB`\n`/mint/data/datafile-0-b` -> `myminio/mc-test-bucket-13776/datafile-0-b`\n`/mint/data/datafile-1-b` -> `myminio/mc-test-bucket-13776/datafile-1-b`\n`/mint/data/datafile-1-kB` -> `myminio/mc-test-bucket-13776/datafile-1-kB`\n`/mint/data/datafile-1.03-MB` -> `myminio/mc-test-bucket-13776/datafile-1.03-MB`\n`/mint/data/datafile-10-MB` -> `myminio/mc-test-bucket-13776/datafile-10-MB`\n`/mint/data/datafile-10-kB` -> `myminio/mc-test-bucket-13776/datafile-10-kB`\n`/mint/data/datafile-100-kB` -> `myminio/mc-test-bucket-13776/datafile-100-kB`\n`/mint/data/datafile-11-MB` -> `myminio/mc-test-bucket-13776/datafile-11-MB`\n`/mint/data/datafile-5243880-b` -> `myminio/mc-test-bucket-13776/datafile-5243880-b`\n`/mint/data/datafile-33-kB` -> `myminio/mc-test-bucket-13776/datafile-33-kB`\n`/mint/data/datafile-5-MB` -> `myminio/mc-test-bucket-13776/datafile-5-MB`\n`/mint/data/datafile-129-MB` -> `myminio/mc-test-bucket-13776/datafile-129-MB`\n`/mint/data/datafile-65-MB` -> `myminio/mc-test-bucket-13776/datafile-65-MB`\n`/mint/data/datafile-6-MB` -> `myminio/mc-test-bucket-13776/datafile-6-MB`\nmc: <ERROR> Failed to copy `/mint/data/datafile-129-MB`. A timeout occurred while trying to lock a resource\nTotal: 0 B, Transferred: 465.91 MiB, Speed: 60.30 KiB/s"
}
(6/15) Running minio-dotnet tests ... FAILED in 26 minutes and 52 seconds
{
  "name": "minio-dotnet : RemoveObject_Test1",
  "function": "Task RemoveObjectAsync(string bucketName, string objectName, CancellationToken cancellationToken = default(CancellationToken))",
  "args": {
    "bucketName": "miniodotnets3wuopgb6cpel2t",
    "objectName": ")]a{{@[@*("
  },
  "duration": 100178,
  "status": "FAIL",
  "alert": "",
  "message": "MinIO API responded with message=Connection error: The operation has timed out.. Status code=0, response=The operation has timed out., content=",
  "error": "Minio.Exceptions.ConnectionException: MinIO API responded with message=Connection error: The operation has timed out.. Status code=0, response=The operation has timed out., content=\n   at Minio.MinioClient.ParseErrorNoContent(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 447\n   at Minio.MinioClient.ParseError(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 431\n   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 605\n   at Minio.MinioClient.ExecuteTaskCoreAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 405\n   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, String uploadId, Int32 partNumber, Byte[] data, Dictionary`2 metaData, Dictionary`2 sseHeaders, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 525\n   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, Stream data, Int64 size, String contentType, Dictionary`2 metaData, ServerSideEncryption sse, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 294\n   at Minio.Functional.Tests.FunctionalTest.RemoveObject_Test1(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2043"
}
(6/15) Running minio-go tests ... FAILED in 10 minutes and 0 seconds
{
  "args": {
    "bucketName": "minio-go-test-e1qvnvxqxud6iqkd",
    "objectName": "tl9vqitio1wnaobpeaisv1pcfgvefq",
    "region": "eu-west-1"
  },
  "duration": 600793,
  "error": "Put http://minio-dev4.minio.io:30170/minio-go-test-e1qvnvxqxud6iqkd/tl9vqitio1wnaobpeaisv1pcfgvefq: net/http: timeout awaiting response headers",
  "function": "MakeBucket(bucketName, region)",
  "message": "PutObject failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(6/15) Running minio-java tests ... FAILED in 15 minutes and 1 seconds
{
  "name": "minio-java",
  "function": "listBuckets()",
  "duration": 900013,
  "status": "FAIL",
  "error": "java.net.SocketTimeoutException: timeout >>> [okio.Okio$4.newTimeoutException(Okio.java:232), okio.AsyncTimeout.exit(AsyncTimeout.java:286), okio.AsyncTimeout$2.read(AsyncTimeout.java:241), okio.RealBufferedSource.indexOf(RealBufferedSource.java:358), okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:230), okhttp3.internal.http1.Http1Codec.readHeaderLine(Http1Codec.java:226), okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:202), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:101), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:264), okhttp3.RealCall.execute(RealCall.java:93), io.minio.MinioClient.execute(MinioClient.java:1097), io.minio.MinioClient.executeGet(MinioClient.java:1311), io.minio.MinioClient.listBuckets(MinioClient.java:3123), FunctionalTest.listBuckets_test(FunctionalTest.java:325), FunctionalTest.runTests(FunctionalTest.java:4011), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(6/15) Running minio-js tests ... FAILED in 11 minutes and 24 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 44,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 72.28.97.53:30170 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16)"
}
(6/15) Running minio-py tests ... FAILED in 6 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "function": "make_bucket(bucket_name, location)",
  "args": {
    "bucket_name": "minio-py-test-a91a599c-63ae-4799-a4eb-4ba20332ce11",
    "location": "default value ('us-east-1')"
  },
  "duration": 6266,
  "message": "HTTPConnectionPool(host='minio-dev4.minio.io', port=30170): Max retries exceeded with url: /minio-py-test-a91a599c-63ae-4799-a4eb-4ba20332ce11/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f04c1497710>: Failed to establish a new connection: [Errno 111] Connection refused',))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 160, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 84, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 74, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 677, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 392, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/lib/python3.6/http/client.py\", line 1264, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1310, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1259, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1038, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 976, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 187, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 172, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f04c1497710>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 188, in test_make_bucket_default_region\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 371, in make_bucket\n    headers=headers)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 336, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 725, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev4.minio.io', port=30170): Max retries exceeded with url: /minio-py-test-a91a599c-63ae-4799-a4eb-4ba20332ce11/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f04c1497710>: Failed to establish a new connection: [Errno 111] Connection refused',))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1968, in main\n    test_make_bucket_default_region(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 196, in test_make_bucket_default_region\n    raise Exception(err)\nException: HTTPConnectionPool(host='minio-dev4.minio.io', port=30170): Max retries exceeded with url: /minio-py-test-a91a599c-63ae-4799-a4eb-4ba20332ce11/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f04c1497710>: Failed to establish a new connection: [Errno 111] Connection refused',))\n",
  "status": "FAIL"
}
(6/15) Running s3cmd tests ... FAILED in 1 seconds
{
  "name": "s3cmd",
  "duration": "335",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-26385/s3cfg mb s3://s3cmd-test-bucket-24363\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.1.0\npython:   3.6.9 (default, Apr 18 2020, 01:56:04) \n[GCC 8.4.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3121, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3030, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 251, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 408, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 1281, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.6/dist-packages/S3/ConnMan.py\", line 251, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.6/http/client.py\", line 948, in connect\n    (self.host,self.port), self.timeout, self.source_address)\n  File \"/usr/lib/python3.6/socket.py\", line 724, in create_connection\n    raise err\n  File \"/usr/lib/python3.6/socket.py\", line 713, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(6/15) Running s3select tests ... FAILED in 12 seconds
{
  "name": "minio-py:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-33317f01-8fe3-4f2a-be45-00df56e6d4ee"
  },
  "duration": 12187,
  "message": "HTTPConnectionPool(host='minio-dev4.minio.io', port=30170): Max retries exceeded with url: /s3select-test-33317f01-8fe3-4f2a-be45-00df56e6d4ee?location= (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe39d54bb70>: Failed to establish a new connection: [Errno 111] Connection refused',))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 160, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 84, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 74, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 677, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 392, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/lib/python3.6/http/client.py\", line 1264, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1310, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1259, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1038, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 976, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 187, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 172, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fe39d54b0b8>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 144, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 371, in make_bucket\n    headers=headers)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 336, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 725, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev4.minio.io', port=30170): Max retries exceeded with url: /s3select-test-33317f01-8fe3-4f2a-be45-00df56e6d4ee/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe39d54b0b8>: Failed to establish a new connection: [Errno 111] Connection refused',))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 183, in test_csv_input_custom_quote_char\n    raise Exception(err)\nException: HTTPConnectionPool(host='minio-dev4.minio.io', port=30170): Max retries exceeded with url: /s3select-test-33317f01-8fe3-4f2a-be45-00df56e6d4ee/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe39d54b0b8>: Failed to establish a new connection: [Errno 111] Connection refused',))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 160, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 84, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 74, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 677, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 392, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/lib/python3.6/http/client.py\", line 1264, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1310, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1259, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1038, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 976, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 187, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 172, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fe39d54bb70>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 186, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1984, in _url_open\n    region = self._get_bucket_region(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1919, in _get_bucket_region\n    region = self._get_bucket_location(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1950, in _get_bucket_location\n    headers=headers)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 336, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 765, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 725, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev4.minio.io', port=30170): Max retries exceeded with url: /s3select-test-33317f01-8fe3-4f2a-be45-00df56e6d4ee?location= (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe39d54bb70>: Failed to establish a new connection: [Errno 111] Connection refused',))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 310, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"./tests.py\", line 188, in test_csv_input_custom_quote_char\n    raise Exception(err)\nException: HTTPConnectionPool(host='minio-dev4.minio.io', port=30170): Max retries exceeded with url: /s3select-test-33317f01-8fe3-4f2a-be45-00df56e6d4ee?location= (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fe39d54bb70>: Failed to establish a new connection: [Errno 111] Connection refused',))\n",
  "status": "FAIL"
}
(6/15) Running security tests ... done in 0 seconds

Executed 6 out of 15 tests successfully.
```
<a name="7223c2e9-27fb-47ea-84a8-0aae6a597c7f"></a>
### 9598-99d6a88/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31812
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6daf399c53e7:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... FAILED in 21 seconds
{
  "name": "awscli",
  "duration": 4510,
  "function": "aws --endpoint-url http://minio-c3.minio.io:31812 s3api get-object --bucket awscli-mint-test-bucket-28667 --key awscli-mint-test-bucket-28667-object /tmp/datafile-0-b\n",
  "status": "FAIL",
  "error": "Etag mismatch for multipart 0 byte object"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 18 seconds
(7/15) Running minio-dotnet tests ... done in 36 minutes and 8 seconds
(8/15) Running minio-go tests ... FAILED in 1 minutes and 0 seconds
{
  "args": {
    "bucketName": "minio-go-test-0u1356p1esgsqm3y",
    "region": "eu-west-1"
  },
  "duration": 60140,
  "error": "Your previous request to create the named bucket succeeded and you already own it.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket(String bucketName)",
  "duration": 60359,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketAlreadyOwnedByYou, message = Your previous request to create the named bucket succeeded and you already own it., bucketName = minio-java-test-11r2sl, objectName = null, resource = /minio-java-test-11r2sl, requestId = 160EDFE522B35EE9, hostId = d6b747fb-6585-4b1a-80e8-44543a857cb2)\nrequest={method=PUT, url=http://minio-c3.minio.io:31812/minio-java-test-11r2sl, headers=Host: minio-c3.minio.io:31812\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200514T105813Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200514/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 374\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.GOGET\nVary: Origin\nX-Amz-Request-Id: 160EDFE522B35EE9\nX-Xss-Protection: 1; mode=block\nDate: Thu, 14 May 2020 10:59:13 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.makeBucket(MinioClient.java:3286), io.minio.MinioClient.makeBucket(MinioClient.java:3199), FunctionalTest.makeBucket_test1(FunctionalTest.java:247), FunctionalTest.runTests(FunctionalTest.java:4005), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(8/15) Running minio-js tests ... FAILED in 12 minutes and 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 660043,
  "status": "FAIL",
  "error": "S3Error: The specified bucket does not exist at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(8/15) Running minio-py tests ... FAILED in 1 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "function": "make_bucket(bucket_name, location)",
  "args": {
    "bucket_name": "minio-py-test-c9dc9568-3887-427d-af48-4de21bf1ac9f",
    "location": "default value ('us-east-1')"
  },
  "duration": 60029,
  "message": "BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 188, in test_make_bucket_default_region\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 374, in make_bucket\n    raise ResponseError(response, method, bucket_name).get_exception()\nminio.error.BucketAlreadyOwnedByYou: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1968, in main\n    test_make_bucket_default_region(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 196, in test_make_bucket_default_region\n    raise Exception(err)\nException: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n",
  "status": "FAIL"
}
(8/15) Running s3cmd tests ... FAILED in 1 minutes and 0 seconds
{
  "name": "s3cmd",
  "duration": "60321",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "ERROR: S3 error: 409 (BucketAlreadyOwnedByYou): Your previous request to create the named bucket succeeded and you already own it."
}
(8/15) Running s3select tests ... FAILED in 7 minutes and 54 seconds
{
  "name": "minio-py:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-35ad2a0f-e895-470c-976b-42ad913cac4c"
  },
  "duration": 473741,
  "message": "BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 144, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 374, in make_bucket\n    raise ResponseError(response, method, bucket_name).get_exception()\nminio.error.BucketAlreadyOwnedByYou: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 310, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"./tests.py\", line 183, in test_csv_input_custom_quote_char\n    raise Exception(err)\nException: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n",
  "status": "FAIL"
}
(8/15) Running security tests ... done in 0 seconds

Executed 8 out of 15 tests successfully.
```
<a name="726b16a7-b612-4ea4-9a50-3aa430b669e1"></a>
### 9598-99d6a88/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32042
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 47e1fb5590cc:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-5eaae1fhweh5r6",
    "objectName": "phii33qpum56mwjp3oz1brdfrtxgww"
  },
  "duration": 571,
  "error": "RequestError: send request failed\ncaused by: Put http://minio-dev7.minio.io:32042/aws-sdk-go-test-5eaae1fhweh5r6/phii33qpum56mwjp3oz1brdfrtxgww?tagging=: EOF",
  "function": "testObjectTagging",
  "message": "AWS SDK Go PUTObjectTagging expected to success but got RequestError",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 2 seconds
(2/15) Running aws-sdk-php tests ... done in 45 seconds
(3/15) Running aws-sdk-ruby tests ... done in 4 seconds
(4/15) Running awscli tests ... done in 1 minutes and 25 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 35 seconds
(7/15) Running minio-dotnet tests ... done in 49 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 49 seconds
{
  "args": {},
  "duration": 92983,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... done in 6 minutes and 57 seconds
(9/15) Running minio-js tests ... done in 52 seconds
(10/15) Running minio-py tests ... done in 4 minutes and 8 seconds
(11/15) Running s3cmd tests ... done in 24 seconds
(12/15) Running s3select tests ... done in 5 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="7ae87232-f3d9-4037-aa83-17b5b34b359e"></a>
### 9598-99d6a88/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:31744
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4756cae449d7:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 12 seconds
(4/15) Running aws-sdk-ruby tests ... done in 14 seconds
(5/15) Running awscli tests ... done in 3 minutes and 44 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 2 minutes and 57 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 20 seconds
(9/15) Running minio-go tests ... done in 3 minutes and 50 seconds
(10/15) Running minio-java tests ... done in 26 minutes and 36 seconds
(11/15) Running minio-js tests ... done in 2 minutes and 37 seconds
(12/15) Running minio-py tests ... FAILED in 49 minutes and 24 seconds
{
  "name": "minio-py:test_thread_safe",
  "function": "put_object(bucket_name, object_name, data, length, content_type, metadata, sse, progress, part_size)",
  "args": {
    "bucket_name": "minio-py-test-d3362d39-7844-4e07-a162-6d6e4b3f6b7e",
    "object_name": "91a282b3-4565-4cfd-a70a-19b1e5359f3d"
  },
  "duration": 2403227,
  "message": "RequestTimeTooSkewed: message: The difference between the request time and the server's time is too large.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1638, in test_thread_safe\n    raise exceptions[0]\nException: RequestTimeTooSkewed: message: The difference between the request time and the server's time is too large.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2128, in main\n    test_thread_safe(client, testfile, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    raise Exception(err)\nException: RequestTimeTooSkewed: message: The difference between the request time and the server's time is too large.\n",
  "status": "FAIL"
}
(12/15) Running s3cmd tests ... done in 1 minutes and 23 seconds
(13/15) Running s3select tests ... done in 22 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="29c4c515-ce50-4d4b-bbf4-5466cc102f24"></a>
### 9598-99d6a88/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31022
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9b6847cbd56d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 3 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 24 seconds
(8/15) Running minio-dotnet tests ... done in 1 hours and 9 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 0 seconds
{
  "args": {
    "bucketName": "minio-go-test-rwls19500224roqx",
    "region": "eu-west-1"
  },
  "duration": 60142,
  "error": "Your previous request to create the named bucket succeeded and you already own it.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 9 minutes and 18 seconds
(10/15) Running minio-js tests ... done in 40 seconds
(11/15) Running minio-py tests ... done in 49 seconds
(12/15) Running s3cmd tests ... done in 13 seconds
(13/15) Running s3select tests ... done in 1 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/fs-v1.go
+++ b/cmd/fs-v1.go
@@ -320,12 +320,6 @@ func (fs *FSObjects) MakeBucketWithLocation(ctx context.Context, bucket, locatio
 		return NotImplemented{}
 	}
 
-	bucketLock := fs.NewNSLock(ctx, bucket, "")
-	if err := bucketLock.GetLock(globalObjectTimeout); err != nil {
-		return err
-	}
-	defer bucketLock.Unlock()
-
 	// Verify if bucket is valid.
 	if s3utils.CheckValidBucketNameStrict(bucket) != nil {
 		return BucketNameInvalid{Bucket: bucket}
@@ -356,12 +350,6 @@ func (fs *FSObjects) MakeBucketWithLocation(ctx context.Context, bucket, locatio
 
 // GetBucketInfo - fetch bucket metadata info.
 func (fs *FSObjects) GetBucketInfo(ctx context.Context, bucket string) (bi BucketInfo, e error) {
-	bucketLock := fs.NewNSLock(ctx, bucket, "")
-	if e := bucketLock.GetRLock(globalObjectTimeout); e != nil {
-		return bi, e
-	}
-	defer bucketLock.RUnlock()
-
 	atomic.AddInt64(&fs.activeIOCount, 1)
 	defer func() {
 		atomic.AddInt64(&fs.activeIOCount, -1)
@@ -438,15 +426,6 @@ func (fs *FSObjects) ListBuckets(ctx context.Context) ([]BucketInfo, error) {
 // DeleteBucket - delete a bucket and all the metadata associated
 // with the bucket including pending multipart, object metadata.
 func (fs *FSObjects) DeleteBucket(ctx context.Context, bucket string, forceDelete bool) error {
-	if !forceDelete {
-		bucketLock := fs.NewNSLock(ctx, bucket, "")
-		if err := bucketLock.GetLock(globalObjectTimeout); err != nil {
-			logger.LogIf(ctx, err)
-			return err
-		}
-		defer bucketLock.Unlock()
-	}
-
 	atomic.AddInt64(&fs.activeIOCount, 1)
 	defer func() {
 		atomic.AddInt64(&fs.activeIOCount, -1)

--- a/cmd/namespace-lock.go
+++ b/cmd/namespace-lock.go
@@ -18,18 +18,17 @@ package cmd
 
 import (
 	"context"
-	"errors"
 	pathutil "path"
 	"runtime"
 	"sort"
 	"strings"
 	"sync"
+	"sync/atomic"
 
 	"fmt"
 	"time"
 
 	"github.com/minio/lsync"
-	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/pkg/dsync"
 )
 
@@ -58,8 +57,8 @@ func newNSLock(isDistXL bool) *nsLockMap {
 
 // nsLock - provides primitives for locking critical namespace regions.
 type nsLock struct {
+	ref uint32
 	*lsync.LRWMutex
-	ref uint
 }
 
 // nsLockMap - namespace lock map, provides primitives to Lock,
@@ -68,7 +67,7 @@ type nsLockMap struct {
 	// Indicates if namespace is part of a distributed setup.
 	isDistXL     bool
 	lockMap      map[string]*nsLock
-	lockMapMutex sync.RWMutex
+	lockMapMutex sync.Mutex
 }
 
 // Lock the namespace resource.
@@ -78,17 +77,16 @@ func (n *nsLockMap) lock(ctx context.Context, volume string, path string, lockSo
 	resource := pathJoin(volume, path)
 
 	n.lockMapMutex.Lock()
-	nsLk, found := n.lockMap[resource]
-	if !found {
-		nsLk = &nsLock{
+	if _, found := n.lockMap[resource]; !found {
+		n.lockMap[resource] = &nsLock{
 			LRWMutex: lsync.NewLRWMutex(ctx),
 			ref:      1,
 		}
-		n.lockMap[resource] = nsLk
 	} else {
 		// Update ref count here to avoid multiple races.
-		nsLk.ref++
+		atomic.AddUint32(&n.lockMap[resource].ref, 1)
 	}
+	nsLk = n.lockMap[resource]
 	n.lockMapMutex.Unlock()
 
 	// Locking here will block (until timeout).
@@ -101,13 +99,12 @@ func (n *nsLockMap) lock(ctx context.Context, volume string, path string, lockSo
 	if !locked { // We failed to get the lock
 
 		// Decrement ref count since we failed to get the lock
-		n.lockMapMutex.Lock()
-		nsLk.ref--
-		if nsLk.ref == 0 {
+		if atomic.AddUint32(&nsLk.ref, ^uint32(0)) == 0 {
 			// Remove from the map if there are no more references.
+			n.lockMapMutex.Lock()
 			delete(n.lockMap, resource)
+			n.lockMapMutex.Unlock()
 		}
-		n.lockMapMutex.Unlock()
 	}
 	return
 }
@@ -115,28 +112,21 @@ func (n *nsLockMap) lock(ctx context.Context, volume string, path string, lockSo
 // Unlock the namespace resource.
 func (n *nsLockMap) unlock(volume string, path string, readLock bool) {
 	resource := pathJoin(volume, path)
-	n.lockMapMutex.RLock()
-	nsLk, found := n.lockMap[resource]
-	n.lockMapMutex.RUnlock()
-	if !found {
+
+	n.lockMapMutex.Lock()
+	defer n.lockMapMutex.Unlock()
+	if _, found := n.lockMap[resource]; !found {
 		return
 	}
 	if readLock {
-		nsLk.RUnlock()
+		n.lockMap[resource].RUnlock()
 	} else {
-		nsLk.Unlock()
+		n.lockMap[resource].Unlock()
 	}
-	n.lockMapMutex.Lock()
-	if nsLk.ref == 0 {
-		logger.LogIf(GlobalContext, errors.New("Namespace reference count cannot be 0"))
-	} else {
-		nsLk.ref--
-		if nsLk.ref == 0 {
-			// Remove from the map if there are no more references.
-			delete(n.lockMap, resource)
-		}
+	if atomic.AddUint32(&n.lockMap[resource].ref, ^uint32(0)) == 0 {
+		// Remove from the map if there are no more references.
+		delete(n.lockMap, resource)
 	}
-	n.lockMapMutex.Unlock()
 }
 
 // dsync's distributed lock instance.
@@ -147,7 +137,7 @@ type distLockInstance struct {
 
 // Lock - block until write lock is taken or timeout has occurred.
 func (di *distLockInstance) GetLock(timeout *dynamicTimeout) (timedOutErr error) {
-	lockSource := getSource()
+	lockSource := getSource(2)
 	start := UTCNow()
 
 	if !di.rwMutex.GetLock(di.opsID, lockSource, timeout.Timeout()) {
@@ -165,7 +155,7 @@ func (di *distLockInstance) Unlock() {
 
 // RLock - block until read lock is taken or timeout has occurred.
 func (di *distLockInstance) GetRLock(timeout *dynamicTimeout) (timedOutErr error) {
-	lockSource := getSource()
+	lockSource := getSource(2)
 	start := UTCNow()
 	if !di.rwMutex.GetRLock(di.opsID, lockSource, timeout.Timeout()) {
 		timeout.LogFailure()
@@ -206,7 +196,7 @@ func (n *nsLockMap) NewNSLock(ctx context.Context, lockersFn func() []dsync.NetL
 
 // Lock - block until write lock is taken or timeout has occurred.
 func (li *localLockInstance) GetLock(timeout *dynamicTimeout) (timedOutErr error) {
-	lockSource := getSource()
+	lockSource := getSource(2)
 	start := UTCNow()
 	readLock := false
 	var success []int
@@ -234,7 +224,7 @@ func (li *localLockInstance) Unlock() {
 
 // RLock - block until read lock is taken or timeout has occurred.
 func (li *localLockInstance) GetRLock(timeout *dynamicTimeout) (timedOutErr error) {
-	lockSource := getSource()
+	lockSource := getSource(2)
 	start := UTCNow()
 	readLock := true
 	var success []int
@@ -260,9 +250,9 @@ func (li *localLockInstance) RUnlock() {
 	}
 }
 
-func getSource() string {
+func getSource(n int) string {
 	var funcName string
-	pc, filename, lineNum, ok := runtime.Caller(2)
+	pc, filename, lineNum, ok := runtime.Caller(n)
 	if ok {
 		filename = pathutil.Base(filename)
 		funcName = strings.TrimPrefix(runtime.FuncForPC(pc).Name(),

--- a/cmd/namespace-lock_test.go
+++ b/cmd/namespace-lock_test.go
@@ -27,7 +27,7 @@ import (
 // position will cause the line number to change and the test to FAIL
 // Tests getSource().
 func TestGetSource(t *testing.T) {
-	currentSource := func() string { return getSource() }
+	currentSource := func() string { return getSource(2) }
 	gotSource := currentSource()
 	// Hard coded line number, 31, in the "expectedSource" value
 	expectedSource := "[namespace-lock_test.go:31:TestGetSource()]"

--- a/cmd/server_test.go
+++ b/cmd/server_test.go
@@ -58,7 +58,7 @@ type check struct {
 // Assert - checks if gotValue is same as expectedValue, if not fails the test.
 func (c *check) Assert(gotValue interface{}, expectedValue interface{}) {
 	if !reflect.DeepEqual(gotValue, expectedValue) {
-		c.Fatalf("Test %s:%s expected %v, got %v", getSource(), c.testType, expectedValue, gotValue)
+		c.Fatalf("Test %s:%s expected %v, got %v", getSource(2), c.testType, expectedValue, gotValue)
 	}
 }
 

--- a/cmd/test-utils_test.go
+++ b/cmd/test-utils_test.go
@@ -57,6 +57,7 @@ import (
 	"github.com/minio/minio-go/v6/pkg/s3utils"
 	"github.com/minio/minio-go/v6/pkg/signer"
 	"github.com/minio/minio/cmd/config"
+	"github.com/minio/minio/cmd/crypto"
 	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/bucket/policy"
@@ -65,6 +66,18 @@ import (
 
 // Tests should initNSLock only once.
 func init() {
+	// disable ENVs which interfere with tests.
+	for _, env := range []string{
+		crypto.EnvAutoEncryptionLegacy,
+		crypto.EnvKMSAutoEncryption,
+		config.EnvAccessKey,
+		config.EnvAccessKeyOld,
+		config.EnvSecretKey,
+		config.EnvSecretKeyOld,
+	} {
+		os.Unsetenv(env)
+	}
+
 	// Set as non-distributed.
 	globalIsDistXL = false
 
@@ -342,27 +355,9 @@ func UnstartedTestServer(t TestErrHandler, instanceType string) TestServer {
 	globalMinioPort = port
 	globalMinioAddr = getEndpointsLocalAddr(testServer.Disks)
 
-	globalConfigSys = NewConfigSys()
+	newAllSubsystems()
 
-	globalIAMSys = NewIAMSys()
-	globalIAMSys.Init(ctx, objLayer)
-
-	buckets, err := objLayer.ListBuckets(ctx)
-	if err != nil {
-		t.Fatalf("Unable to list buckets on backend %s", err)
-	}
-
-	globalPolicySys = NewPolicySys()
-	globalPolicySys.Init(buckets, objLayer)
-
-	globalNotificationSys = NewNotificationSys(testServer.Disks)
-	globalNotificationSys.Init(buckets, objLayer)
-
-	globalLifecycleSys = NewLifecycleSys()
-	globalLifecycleSys.Init(buckets, objLayer)
-
-	globalBucketSSEConfigSys = NewBucketSSEConfigSys()
-	globalBucketSSEConfigSys.Init(buckets, objLayer)
+	initAllSubsystems(objLayer)
 
 	return testServer
 }

